scripts/auto_lib.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+"""Wraps modules to forward to content or background automatically
+
+"""
+
+import os
+import sys
+
+WRAPPER_FILE = '''// AUTOGENERATED FROM {filename} BY auto_lib.py
+//
+// Automatically forwards invocations to the appropriate context (content
+// script, background script, commandline iframe)
+//
+// DO NOT EDIT
+
+import * as excmd_lib from "@src/lib/excmd_lib"
+import * as _excmds from "@{filename_minus_extension}"
+export default excmd_lib.forwardedTo{source_cap}("{messagetype}", _excmds)
+'''
+
+MESSAGE_NAMES_HEADER = '''// AUTOGENERATED BY auto_lib.py
+//
+// Autogenerated message names for transparent context-forwarding wrappers
+// created by auto_lib.py. Each of these corresponds to the wrapper for a
+// single module.
+//
+// DO NOT EDIT
+'''
+
+MESSAGE_NAMES_FILE_PATH = "src/lib/.auto_lib_wrapper_names.generated.ts"
+
+def MangleFilename(source, filename):
+    return filename.replace(source + '_lib', 'lib/generated')
+
+def AutoWrapperMessageName(source, filename):
+    return "auto_lib_wrapper:{}->{}".format(filename, source)
+
+def Wrap(source, filename):
+    output_filename = MangleFilename(source, filename)
+    message_name = AutoWrapperMessageName(source, filename)
+    if os.path.exists(output_filename):
+        with open(output_filename, 'r') as f:
+            firstline = f.readline()
+            if not firstline.startswith("// AUTOGENERATED"):
+                raise ValueError("File {} is not autogenerated and cannot be clobbered".format(output_filename))
```

Comment:
This error is unlikely to ever be fired because EnsureGeneratedDirExistsAndIsEmpty() deletes everything in the dir.

If you consider not clobbering to be an important property, check before you delete. Else consider removing this check to avoid giving the illusion of safety.